JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java

```diff
@@ -78,35 +80,52 @@ public void setKBClient(TripleStoreClientInterface kbClient) {
+	 *             If not specified, default value: TimeSeries
+	 * @param duration Required for Average Time Series. Numeric duration of the averaging period for Average TimeSeries of type Duration. Only positive values are allowed. (optional)
+	 * @param unit Required for Average Time Series. Temporal unit type of the averaging period for Average TimeSeries. (optional)
+	 *             Allowed values of type ChronoUnit:
+	 *             ChronoUnit.SECONDS, ChronoUnit.MINUTES, ChronoUnit.HOURS, ChronoUit.DAYS, ChronoUnit.WEEKS, ChronoUnit.MONTHS, ChronoUnit.YEARS
+	 *
+	 */
+	public void initTimeSeries(List<String> dataIRIs, List<Class<?>> dataClass, String timeUnit, Connection conn, String type, Duration duration, ChronoUnit unit) {
+		String tsIRI;
+		// Create random time series IRI in the format: <Namespace><ClassName>_<UUID>
+		if (type.equals(null)){
+			tsIRI = TimeSeriesSparql.ns_kb + "Timeseries_" + UUID.randomUUID();
+		}
+		else if(type.equals(TimeSeriesSparql.StepwiseCumulative)){
+			tsIRI = TimeSeriesSparql.StepwiseCumulative + "Timeseries_" + UUID.randomUUID();
+		}
+		else if(type.equals(TimeSeriesSparql.CumulativeTotal)){
+			tsIRI = TimeSeriesSparql.CumulativeTotal+  "Timeseries_" + UUID.randomUUID();
+		}
+		else if(type.equals(TimeSeriesSparql.Instantaneous)){
+			tsIRI = TimeSeriesSparql.Instantaneous + "Timeseries_" + UUID.randomUUID();
+		}
+		else if(type.equals(TimeSeriesSparql.Average)){
+			tsIRI = TimeSeriesSparql.Average + "Timeseries_" + UUID.randomUUID();
+		}
+		else {
+			throw new JPSRuntimeException(exceptionPrefix + "TimeSeries type: " + type + " is invalid");
+		}
 
-    	// Create random time series IRI in the format: <Namespace><ClassName>_<UUID>
-    	String tsIRI = TimeSeriesSparql.ns_kb + "Timeseries_" + UUID.randomUUID();
-    	
-    	// Step1: Initialise time series in knowledge base
-    	// In case any exception occurs, nothing will be created in kb, since JPSRuntimeException will be thrown before 
-    	// interacting with triple store and SPARQL query is either executed fully or not at all (no partial execution possible)
-		
-		// Obtain RDB URL from connection object, exception thrown when connection is down
-		String rdbURL;
+		// Step1: Initialise time series in knowledge base
+		// In case any exception occurs, nothing will be created in kb, since JPSRuntimeException will be thrown before
+		// interacting with triple store and SPARQL query is either executed fully or not at all (no partial execution possible)
 		try {
-			rdbURL = conn.getMetaData().getURL();
-		} catch (SQLException e) {
-			// this ensures rdfClient.initTS always has a valid string
-			LOGGER.warn(e.getMessage());
-			LOGGER.warn("Failed to get RDB URL from connection object, setting RDB URL to = \"\"");
-			rdbURL = "";
-		}
-		
-   		try {
-   			rdfClient.initTS(tsIRI, dataIRIs, rdbURL, timeUnit);
+			rdfClient.initTS(tsIRI, dataIRIs, conn.getMetaData().getURL(), timeUnit, type, duration, unit);
```

Comment:
This should use rdbURL
`try {
			rdbURL = conn.getMetaData().getURL();
		} catch (SQLException e) {
			// this ensures rdfClient.initTS always has a valid string
			LOGGER.warn(e.getMessage());
			LOGGER.warn("Failed to get RDB URL from connection object, setting RDB URL to = \"\"");
			rdbURL = "";
		}`

Reply:
Have a look at the "deprecated" initTimeSeries